### PR TITLE
Add callbacks "onIndexChange" and "onVisibleChange" to PhotoProvider

### PR DESCRIPTION
**Pull request by CalebBarnes**
These callbacks make it much easier to integrate the slider with custom components and state.

### CODE REVIEW

**Comment by CalebBarnes** [comment]
For context I use react-slick with Gatsby and gatsby-image (custom image component) and then when the user clicks on the image in the slick slider it opens the react-photo-view slider.  I want these callback functions to be able to keep the slick slider in sync with the react-photo-view slider.

File: README.md [hunk 1]
```diff
@@ -84,6 +84,10 @@ function ImageView() {
 | imageClassName | string                            | 否   | 图片 className             |
 | loadingElement | JSX.Element                       | 否   | 自定义 loading             |
 | brokenElement  | JSX.Element &#124; ((photoProps: brokenElementDataType) => JSX.Element)                       | 否   | 加载失败 Element           |
+| afterIndexChange | (index, state) => void;         | 否   | 索引改变回调                |
```
Comment (MinJieLiu) [inline README.md:87]: afterIndexChange => onIndexChange
与 PhotoSlider 的 onIndexChange 保持一致

File: README.md [hunk 2]
```diff
@@ -84,6 +84,10 @@ function ImageView() {
...
+| afterShow | (index, state) => void;                | 否   | 滑块显示回调                |
+| afterClose | (index, state) => void;               | 否   | 滑块关闭回调                |
```
Comment (MinJieLiu) [inline README.md:89]: 这两个API 可以合并成一个 onVisibleChange
(visible: boolean, index: number, state) => void

Comment (CalebBarnes) [reply README.md:89]: Yes that's a good idea.

File: README.md [hunk 3]
```diff
@@ -69,25 +69,23 @@ function ImageView() {
...
+| toolbarRender   | (overlayProps) => React.ReactNode                                       | 否   | 工具栏渲染                 |
+| className       | string                                                                  | 否   | className                  |
+| maskClassName   | string                                                                  | 否   | 遮罩 className             |
+| viewClassName   | string                                                                  | 否   | 图片容器 className         |
+| imageClassName  | string                                                                  | 否   | 图片 className             |
+| loadingElement  | JSX.Element                                                             | 否   | 自定义 loading             |
+| brokenElement   | JSX.Element &#124; ((photoProps: brokenElementDataType) => JSX.Element) | 否   | 加载失败 Element           |
+| onIndexChange   | (index:number, state: PhotoProviderState) => void;                      | 否   | 索引改变回调               |
+| onVisibleChange | (visible: boolean, index: number, state: PhotoProviderState) => void;   | 否   | 可见性更改回调             |
```
Comment (CalebBarnes) [inline README.md:88]: I don't know Chinese so I just used google translate.

**Comment by MinJieLiu** [comment]
0.5.4

**Pull request merged**